vic/drivers/classic/src/write_model_state.c

```diff
@@ -293,10 +298,19 @@ write_model_state(all_vars_struct *all_vars,
                            filep->statefile);
                 }
                 else {
-                    fprintf(filep->statefile, " %f", energy[veg][band].T[nidx]);
+                    fprintf(filep->statefile, " "ASCII_STATE_FLOAT_FMT, energy[veg][band].T[nidx]);
                 }
             }
 
+            /* Write foliage temperature */
```

Comment:
Need to also increment the Nbytes count to include the size of Tfoliage, on line 103 (I can't seem to add an inline comment to that line of this diff, so I'm adding the note here).  You can look at how the other variables are done and pattern Tfoliage after them.  It should be a simple 1-liner.